scripts/utils.sh

```diff
@@ -198,27 +198,45 @@ function find_tmp_dir() {
   echo ${tmp}
 }
 
+# encode API token
+# $1 = api token
+encode_api_token(){
+  local api_token=${1}
+  # ideally we would use base64's option -w but it's not available in all envs
+  echo -n "$api_token" | base64 | tr -d \\n
+}
+
 # $1 = optional: endpoint, defaults to http://localhost:3001
 get_native_address(){
-  local endpoint=${1:-localhost:3001}
-  local cmd="curl --silent --max-time 5 ${endpoint}/api/v2/account/addresses"
+  local with_token_endpoint=${1:-localhost:3001}
+  # split string into api token and endpoint
+  local api_token=${with_token_endpoint%\@*}
+  local endpoint=${with_token_endpoint#*\@}
+  local api_token_encoded=$(encode_api_token $api_token)
 
-  # try every 5 seconds for 5 minutes
-  local result
-  result=$(try_cmd "${cmd}" 30 5)
+  local cmd="curl --silent --max-time 5 ${endpoint}/api/v2/account/addresses"
+  if [[ -n $api_token ]]; then
+    cmd="$cmd --header 'Authorization: Basic ${api_token_encoded}'"
+  fi
 
+  local result=$(eval "$cmd")
```

Comment:
Why did you remove the retry here?

Reply:
I spend quite sometime getting the retry to work with that `cmd` string, it fails for some reason on that particular string.
I have in the description of this PR a task to open a new issue to fix this.